podcastfy/core/podcast.py

```diff
@@ -1,64 +1,17 @@
 from abc import ABC, abstractmethod
 from enum import Enum
-from typing import List, Optional, Dict, Any, Union, Callable
 from pathlib import Path
+from typing import List, Optional, Dict, Any, Callable
 from pydub import AudioSegment as PydubAudioSegment
-from urllib.parse import urlparse
 from functools import wraps
 from contextlib import contextmanager
 
 
 class PodcastState(Enum):
     INITIALIZED = 0
-    CONTENT_PARSED = 1
-    TRANSCRIPT_BUILT = 2
-    AUDIO_SEGMENTS_BUILT = 3
-    ASSEMBLED = 4
-
-
-class ParserBackend(ABC):
-    @abstractmethod
-    def parse(self, url: str) -> str:
-        pass
-
-
-class WebsiteParser(ParserBackend):
-    def parse(self, url: str) -> str:
-        # Implementation for parsing website content
-        return f"Parsed content from {url}"
-
-
-class PDFParser(ParserBackend):
-    def parse(self, url: str) -> str:
-        # Implementation for parsing PDF content
-        return f"Parsed content from PDF {url}"
-
-
-class YouTubeParser(ParserBackend):
-    def parse(self, url: str) -> str:
-        # Implementation for parsing YouTube video transcripts
-        return f"Parsed transcript from YouTube video {url}"
-
-
-class ParserCollection:
-    def __init__(self):
-        self.parsers = {
-            'http': WebsiteParser(),
-            'https': WebsiteParser(),
-            'pdf': PDFParser(),
-            'youtube': YouTubeParser()
-        }
-
-    def parse(self, url: str) -> str:
-        parsed_url = urlparse(url)
-        if parsed_url.scheme in ['http', 'https']:
-            if 'youtube.com' in parsed_url.netloc or 'youtu.be' in parsed_url.netloc:
-                return self.parsers['youtube'].parse(url)
-            return self.parsers[parsed_url.scheme].parse(url)
-        elif url.lower().endswith('.pdf'):
-            return self.parsers['pdf'].parse(url)
-        else:
-            raise ValueError(f"Unsupported URL format: {url}")
+    TRANSCRIPT_BUILT = 1
+    AUDIO_SEGMENTS_BUILT = 2
+    ASSEMBLED = 3
```

Comment:
Consider adding in-line comments to explain what each static value represents, e.g. What's 'ASSEMBLED'?